components/datetime/src/provider/date_time.rs

```diff
@@ -309,21 +311,38 @@ where
     }
 }
 
+/// Internal enum to represent the kinds of month symbols for interpolation
+pub enum MonthPlaceholderValue<'a> {
+    PlainString(&'a str),
+    StringNeedingLeapPrefix(&'a str),
```

Comment:
thought: not a fan of still needing this; would like to see if we can move to a more scoped hack that uses NumericPattern

also does this need to be fully `pub`?

Reply:
It can be `pub(crate)`.

`StringNeedingLeapPrefix` is there for the legacy code and I think it will be dead (never constructed) once we delete the legacy code

Reply:
Yeah please make it `pub(crate)`.